test/connection_switching_test.rb

```diff
@@ -519,11 +519,20 @@ class ShardedModel < ActiveRecord::Base
         end
 
         it "will :include things via has_and_belongs associations correctly" do
+          if ActiveRecord::VERSION::MAJOR >= 4 && ActiveRecord::VERSION::MINOR >= 2
+            skip("There's a bug with Rails 4.2")
+          end
           a = Account.where(:id => 1001).includes(:people).first
           assert a.people.size > 0
           assert_equal 'slave person', a.people.first.name
         end
 
+        it "sets up has and belongs to many sharded-ness correctly" do
+          if Object.const_defined?('Account::HABTM_People')
```

Comment:
Wouldn't it be better to check on ActiveRecord version here? This test could be randomly skipped and we'd never know.